src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -2921,15 +2891,15 @@ private void RemoveDirectoryInfoItem(DirectoryInfo directory, bool recurse, bool
                     return;
                 }
 
-                bool isDirectory;
-                Exception accessException;
-
-                if (!Utils.NativeItemExists(directory.FullName, out isDirectory, out accessException))
+                try
                 {
-                    return;
+                    if (!Utils.ItemExists(directory.FullName, out bool unused))
+                    {
+                        // Directory does not exist
+                        return;
+                    }
                 }
-
-                if (accessException != null)
+                catch (Exception accessException)
                 {
                     ErrorRecord errorRecord = new ErrorRecord(accessException, "RemoveFileSystemItemUnAuthorizedAccess", ErrorCategory.PermissionDenied, directory);
```

Comment:
Looks to me it's expecting an `UnauthorizedAccessException`, shall we narrow the catch to only catch this type of exception?

Reply:
Here I only preserve the logic.
Really we should clean up the code too because .Net Core support reparse points and recursion https://github.com/dotnet/corefx/blob/f25eb288a449010574a6e95fe298f3ad880ada1e/src/System.IO.FileSystem/src/System/IO/FileSystem.Windows.cs#L359

Seems we should clean up all code in FileSystemProvider - perhaps we don't need FileSystemProviderV2 :-)

Reply:
@daxian-dbw I opened #7017 to clean up the code.

Reply:
I'm fine leaving this catch as is.